internal/scaffold/helm/watches.go

```diff
@@ -43,6 +54,44 @@ func (s *WatchesYAML) GetInput() (input.Input, error) {
 	return s.Input, nil
 }
 
+func UpdateHelmWatchForResource(r *scaffold.Resource, absProjectPath string) error {
+
+	watchFilePath := filepath.Join(absProjectPath, WatchesYamlFile)
+	watchYAML, err := ioutil.ReadFile(watchFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read watch manifest %v: %v", watchFilePath, err)
+	}
+	watchList := []watches.Watch{}
+	err = yaml.Unmarshal(watchYAML, &watchList)
+	if err != nil {
+		log.Error(err, "Failed to unmarshal watch config")
+		return nil
+	}
+	gvk := schema.GroupVersionKind{
+		Version: r.Version,
+		Group:   r.FullGroup,
+		Kind:    r.Kind,
+	}
+	for _, watch := range watchList {
+		if watch.GroupVersionKind == gvk {
+			// dupe detected
+			dupeWatch = true
+			return fmt.Errorf("duplicate GVK: %v", watch.GroupVersionKind.String())
+		}
+	}
+	if dupeWatch == false {
```

Comment:
If you've gotten to this line, we _know_ dupeWatch is false, because we return an error as soon as we find a dupe. Maybe we don't even need `dupeWatch`?